### PR TITLE
cypress cache list prints last accessed date

### LINKED ISSUE #6682: Print last accessed date in command "cypress cache list"
Related to #6404

- when doing `npx cypress cache list` should print each cached binary version and when that version was used on that machine the last time (today, a day ago, a month). 

This allows the user to clean or prune the cache intelligently removing Cypress versions that are no longer needed

### PR DESCRIPTION
**Pull request by bahmutov**
- Closes #6682

### User facing changelog

- `cypress cache list` now prints the last time the cached binary was accessed for each version present.

### Additional details

The user can see which old(er) versions of Cypress binary can be deleted

### How has the user experience changed?

Screen Shot 2020-03-09 at 12 19 16 PM

Note: HTML of the test snapshots after passing just text snapshot check is saved on CircleCI as a static artifact. See example in https://271408-31629751-gh.circle-artifacts.com/0/cli/test/html/list-of-versions.html

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue been tagged with a release in ZenHub? 
- [x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? https://github.com/cypress-io/cypress-documentation/pull/2598

## Why no folder size?

Seems cross-platform fast folder size is hard, so let's start with last access timestamp and CLI HTML test artifacts, then we can add one more column with folder size

### CODE REVIEW

File: cli/lib/tasks/cache.js [hunk 1]
```diff
@@ -2,7 +2,19 @@ const state = require('./state')
...
+const Table = require('cli-table3')
+const moment = require('moment')
+const chalk = require('chalk')
 
+// color for numbers and show values
+const g = chalk.green
+// last used color
+const u = chalk.cyan
+// accepts and table headings
+const a = chalk.red
```
Comment (jennifer-shehane) [inline cli/lib/tasks/cache.js]: These const names are more confusing than helpful. If I saw `a('text')`, I would have zero idea what `a` means or what it is doing. If the functionality is just coloring text red, something like `red('text')` makes more sense. Clarity > brevity.

File: cli/lib/tasks/cache.js [hunk 2]
```diff
@@ -15,15 +27,51 @@ const clear = () => {
 
 const list = () => {
   return getCachedVersions()
-  .then((versions) => {
-    logger.log(versions.join(', '))
+  .then((binaries) => {
+    const table = new Table({
+      head: [a('version'), a('last used')],
```
Comment (jennifer-shehane) [inline cli/lib/tasks/cache.js]: I would probably opt for white (I think this is default) or grey text here. The red does not look good visually 😅. Every other use case of `chalk.red` that I found also denotes errors/warnings/failures so seems inappropriate functionality-wise.

File: cli/lib/tasks/cache.js [hunk 3]
```diff
@@ -15,15 +27,51 @@ const clear = () => {
...
+    })
+
+    binaries.forEach((x) => {
+      const versionString = g(x.version)
+      const lastUsed = x.accessed ? u(x.accessed) : 'unknown'
```
Comment (jennifer-shehane) [inline cli/lib/tasks/cache.js]: Again, what is `x` here? This variable name makes this method harder to read. Suggest calling it `binary` if it is a singular of `binaries. Clarity > brevity.

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
I thought the original scope of this was to list the directory the cache is located in and also the file size of the binary. Has that been removed from scope or is this just partially addressing #6404? (which is fine if it is!)

**Comment by bahmutov** [comment]
@jennifer-shehane thanks for good code review, appreciate it. I have renamed everything for clarity.

For the folder sizes - I left a note in the PR description. Turns out Node's recursive folder size is _much_ slower than native "du -h" on Mac for example. So I think let's start with this table, since it already introduces better UI and a feature (last access timestamp) plus CLI visual artifacts. Then we can reconsider how to implement fast folder size across platforms. I am thinking a separate NPM tool that tries to use native tools with Node's fallback

**Review by jennifer-shehane — APPROVED** [review]
Looks good - tested manually as well. I would still like the path to each cache printed because I have to run `cypress cache path` directly after to see where these are. 

@bahmutov I would suggest opening a new issue for the directory location and size if you intend to close the original issue from this PR. 

Screen Shot 2020-03-09 at 12 19 16 PM

**Pull request merged**